feat: Profile has assets on other chains

Jarsen136 opened the pull request:
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

  👇 __ Let's make a quick check before the contribution.

  ## PR Type

  - [x] Feature

  ## Needs Design check

  - @exezbcz please review

  ## Context

  - [x] Closes #8742

  #### Did your issue had any of the "$" label on it?

  - [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

  ## Screenshot 📸

  - [x] My fix has changed UI

It has nft/collection on any other chains.
<img width="1557" alt="image" src="https://github.com/kodadot/nft-gallery/assets/31397967/81585df6-3862-4181-83f1-f91d8e8cc234">

<img width="1437" alt="image" src="https://github.com/kodadot/nft-gallery/assets/31397967/59cdc73f-8edf-4f4e-bf9b-aa0c27a59b76">




It have no collection on any other chains.
<img width="1545" alt="image" src="https://github.com/kodadot/nft-gallery/assets/31397967/587e06ad-4025-4f25-8fc5-3bcbed8ad963">


  ## Copilot Summary
  copilot:summary

  copilot:poem

exezbcz commented:
merry xmas - the (4) there is indicator of how much networks the user has art on 

- also the cta should be to a chain where the user has any art 

>it will be displayed whenever there are no assets on the current chain, but are on others

- then, if it is on more chains than one (for example ahp and RMRK2) it shows a redirect based on the hierarchy mentioned - in this case ahp

Jarsen136 commented:
> merry xmas - the (4) there is indicator of how much networks the user has art on
> 
> * also the cta should be to a chain where the user has any art
> 
> > it will be displayed whenever there are no assets on the current chain, but are on others
> 
> * then, if it is on more chains than one (for example ahp and RMRK2) it shows a redirect based on the hierarchy mentioned - in this case ahp

✅ Done. Could you please check again? @exezbcz

exezbcz commented:
can you please use this message for the empty state 
![image](https://github.com/kodadot/nft-gallery/assets/90852205/bb57fd9e-1606-4cdc-8501-0af76fd7c9f9)

- when a user does not have anything on the chain not even on others 


otherwise, it's working nicely! thanks

Jarsen136 commented:
> can you please use this message for the empty state ![image](https://private-user-images.githubusercontent.com/90852205/292851937-bb57fd9e-1606-4cdc-8501-0af76fd7c9f9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM1OTU5OTcsIm5iZiI6MTcwMzU5NTY5NywicGF0aCI6Ii85MDg1MjIwNS8yOTI4NTE5MzctYmI1N2ZkOWUtMTYwNi00Y2RjLTg1MDEtMGFmNzZmZDdjOWY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI2VDEzMDEzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZmMWIyZjcxZjhjZWUwY2FiODk4ZTE2ZTZmM2IwYWQ3MTg2ZDEwNGY1MTkxMmIyZWFkNTQ0MmUzNGE5OWEzOTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Gy__m3HudILIrA_nn-SlKCzX85Q42rgl2mZ-KpOjt-E)
> 
> * when a user does not have anything on the chain not even on others
> 
> otherwise, it's working nicely! thanks

✅ Yes, I have updated it.

prury commented:
suggestions:

perhaps add skeleton and then show one of the two messages?

https://github.com/kodadot/nft-gallery/assets/36627808/56a0858a-0c8f-4e56-85d4-325887bab311

provide better chain naming instead of AHP or KSM, but only if easy to implement

roiLeo commented on components/profile/ProfileDetail.vue:
```diff
@@ -199,6 +211,9 @@ import { chainsWithMintInteraction } from '@/composables/collectionActivity/help
 import CollectionFilter from './CollectionFilter.vue'
 import GridLayoutControls from '@/components/shared/GridLayoutControls.vue'
+import ProfileEmptyResult from '@/components/profile/ProfileEmptyResult.vue'
+import { CHAINS, type Prefix } from '@kodadot1/static'
```
should we use `CHAINS` or `chainNames`?

Jarsen136 replied:
It's not related to the chain name here.

roiLeo commented on components/profile/ProfileDetail.vue:
```diff
@@ -349,6 +367,72 @@ useAsyncData('tabs-count', async () => {
   }
 })
 
+const fetchTabsCount = async (chain: Prefix) => {
```
add more query on profile?
consider refactor since +100 LOC

Jarsen136 replied:
It would fetch asset data from different network on profile page.
This function has only 40 LOC. There is no `big` function defined in this file.

roiLeo commented on locales/en.json:
```diff
@@ -837,6 +837,9 @@
     "holdings": "Holdings",
     "gains": "Gains",
     "showQr": "Show QR",
+    "searchNoResultsTitle": "No Assets On Selected Chain",
+    "searchNoResultsText": "Explore The Artist's Protfolio On Other Networks ({0})",
+    "searchNoResultsButton": "Explore On {0}",
```
why duplicated key?

Jarsen136 replied:
They are in different scopes and they are used in different places. 
`general. searchNoResultsTitle ` and `profile.searchNoResultsTitle`

roiLeo replied:
I don't understand, there is no point of having duplicate text in translation file even if "different" scopes

Jarsen136 replied:
Oh never mind. I forget to delete the similar key in this file : ) my bad

roiLeo commented on components/profile/ProfileDetail.vue:
```diff
@@ -199,6 +211,9 @@ import { chainsWithMintInteraction } from '@/composables/collectionActivity/help
 import { Interaction } from '@kodadot1/minimark/v1'
 import CollectionFilter from './CollectionFilter.vue'
 import GridLayoutControls from '@/components/shared/GridLayoutControls.vue'
+import ProfileEmptyResult from '@/components/profile/ProfileEmptyResult.vue'
```
use auto import instead

Jarsen136 replied:
updated

Jarsen136 commented:
> suggestions:
> 
> perhaps add skeleton and then show one of the two messages?
> 
>  skel.mp4 
> provide better chain naming instead of AHP or KSM, but only if easy to implement
> 
> edit: functionality working nicely tho

✅DONE
 I have added a loading skeleton and updated the chain name for it.

roiLeo commented on components/profile/ProfileEmptyResult.vue:
```diff
@@ -0,0 +1,36 @@
+import { type Prefix, chainNames } from '@kodadot1/static'
+
+const props = defineProps({
+  hasAssetPrefixList: { type: Array<Prefix>, default: () => [] },
```
prefix `has` means boolean to me, maybe change it to something more meaningfull?

Jarsen136 replied:
Ok. I changed it to another naming

roiLeo commented on components/profile/ProfileEmptyResult.vue:
```diff
@@ -0,0 +1,36 @@
+const { redirectAfterChainChange } = useChainRedirect()
+const { setUrlPrefix } = usePrefix()
+const totalChainsLength = computed(() => props.hasAssetPrefixList.length)
+const nextPrefix = computed(() => {
+  return props.hasAssetPrefixList[0]
+})
```
what if empty?
```suggestion
const nextPrefix = computed(() =>  props.hasAssetPrefixList[0])
```

Jarsen136 replied:
remove blank
```suggestion
const nextPrefix = computed(() => props.hasAssetPrefixList[0])
```

Jarsen136 replied:
> what if empty?

If it's empty, this component will not be rendered. so, it would not happen : )

prury commented on locales/en.json:
```diff
@@ -837,6 +837,9 @@
     "gains": "Gains",
     "showQr": "Show QR",
+    "searchNoResultsTitle": "No Assets On Selected Chain",
+    "searchNoResultsText": "Explore The Artist's Protfolio On Other Networks ({0})",
```
```suggestion
    "searchNoResultsText": "Explore The Artist's Portfolio On Other Networks ({0})",
```

roiLeo replied:
remove "The"?
btw why everything is capitalize 😭 ?

exezbcz replied:
we can make it normal - its from the figma text style i suppose

Jarsen136 replied:
> remove "The"?
OK

> btw why everything is capitalize 😭 ?
It's just another solution to apply `capitalize` css style.

Jarsen136 replied:
> we can make it normal - its from the figma text style i suppose

Alright

yangwao commented:
Thanks!
pay 50 usd

yangwao commented:
😍 Perfect, I’ve sent the payout 
💵 $50 @ 8.64 USD/DOT ~ 5.787 $DOT
🧗 [16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z](https://polkadot.subscan.io/account/16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)
🔗 [0xe5c965e471b951ab566018e0bf4f7ca3666f1936b89fe482aaa30fad6424d9c4](https://polkadot.subscan.io/extrinsic/0xe5c965e471b951ab566018e0bf4f7ca3666f1936b89fe482aaa30fad6424d9c4)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)
